test/test_cases.cpp

```diff
@@ -0,0 +1,1283 @@
+  EXPECT_EQ("97", uut_obj[0]);
+  EXPECT_EQ("98", uut_obj[1]);
+  EXPECT_EQ("99", uut_obj[2]);
+
+  ASSERT_EQ(3, source_vector_pod.size());
+  ASSERT_EQ(3, source_vector_obj.size());
+}
+
+TEST_F(ConstructorTests, MoveConstructorNoAllocator) {
+  ekuvector<int32_t> source_vector_pod({97, 98, 99});
+  ekuvector<std::string> source_vector_obj({"97", "98", "99"});
+
+  ekuvector<int32_t> uut_pod(std::move(source_vector_pod));
+  ekuvector<std::string> uut_obj(std::move(source_vector_obj));
+
+  ASSERT_EQ(3, uut_pod.size());
+  EXPECT_EQ(97, uut_pod[0]);
+  EXPECT_EQ(98, uut_pod[1]);
+  EXPECT_EQ(99, uut_pod[2]);
+
+  ASSERT_EQ(3, uut_obj.size());
+  EXPECT_EQ("97", uut_obj[0]);
+  EXPECT_EQ("98", uut_obj[1]);
+  EXPECT_EQ("99", uut_obj[2]);
+
+  // sources should have been emptied
+  ASSERT_EQ(0, source_vector_pod.size());
+  ASSERT_EQ(0, source_vector_obj.size());
+}
+
+TEST_F(ConstructorTests, ConstructFromInitiliazerListNoAllocator) {
+  std::initializer_list<int32_t> source_init_list_pod({97, 98, 99});
+  std::initializer_list<std::string> source_init_list_obj({"97", "98", "99"});
+
+  ekuvector<int32_t> uut_pod(source_init_list_pod);
+  ekuvector<std::string> uut_obj(source_init_list_obj);
+
+  ASSERT_EQ(3, uut_pod.size());
+  EXPECT_EQ(97, uut_pod[0]);
+  EXPECT_EQ(98, uut_pod[1]);
+  EXPECT_EQ(99, uut_pod[2]);
+
+  ASSERT_EQ(3, uut_obj.size());
+  EXPECT_EQ("97", uut_obj[0]);
+  EXPECT_EQ("98", uut_obj[1]);
+  EXPECT_EQ("99", uut_obj[2]);
+}
+
+class AssignmentTests : public EkuVectorTests {};
+
+TEST_F(AssignmentTests, CopyAssignment) {
+  ekuvector<int32_t> source_vector_pod({97, 98, 99});
+  ekuvector<std::string> source_vector_obj({"97", "98", "99"});
+
+  ekuvector<int32_t> uut_pod;
+  ekuvector<std::string> uut_obj;
+
+  ASSERT_EQ(0, uut_pod.size());
+  ASSERT_EQ(0, uut_obj.size());
+
+  uut_pod = source_vector_pod;
+  uut_obj = source_vector_obj;
+
+  ASSERT_EQ(3, source_vector_pod.size());
+  ASSERT_EQ(3, source_vector_obj.size());
+
+  ASSERT_EQ(3, uut_pod.size());
+  EXPECT_EQ(97, uut_pod[0]);
+  EXPECT_EQ(98, uut_pod[1]);
+  EXPECT_EQ(99, uut_pod[2]);
+
+  ASSERT_EQ(3, uut_obj.size());
+  EXPECT_EQ("97", uut_obj[0]);
+  EXPECT_EQ("98", uut_obj[1]);
+  EXPECT_EQ("99", uut_obj[2]);
+}
+
+TEST_F(AssignmentTests, MoveAssignment) {
+  ekuvector<int32_t> source_vector_pod({97, 98, 99});
+  ekuvector<std::string> source_vector_obj({"97", "98", "99"});
+
+  ekuvector<int32_t> uut_pod;
+  ekuvector<std::string> uut_obj;
+
+  ASSERT_EQ(0, uut_pod.size());
+  ASSERT_EQ(0, uut_obj.size());
+
+  uut_pod = std::move(source_vector_pod);
+  uut_obj = std::move(source_vector_obj);
+
+  ASSERT_EQ(3, uut_pod.size());
+  EXPECT_EQ(97, uut_pod[0]);
+  EXPECT_EQ(98, uut_pod[1]);
+  EXPECT_EQ(99, uut_pod[2]);
+
+  ASSERT_EQ(3, uut_obj.size());
+  EXPECT_EQ("97", uut_obj[0]);
+  EXPECT_EQ("98", uut_obj[1]);
+  EXPECT_EQ("99", uut_obj[2]);
+
+  // Note that the current implementation of this operator overload
+  // does not empty the source vectors
```

Comment:
@glpuga shall we test a move-construction actually occurred? Beware of [SSO](https://stackoverflow.com/questions/21694302/what-are-the-mechanics-of-short-string-optimization-in-libc).

Reply:
I addressed this comment here: https://github.com/glpuga/ekucontainers/pull/1/commits/ff1bd82d6f3f83f0af4d0261fff481a58c1076a2